pallets/schemas/src/tests/other_tests.rs

```diff
@@ -698,3 +703,59 @@ fn create_schema_with_append_only_setting_and_non_itemized_should_fail() {
 		);
 	})
 }
+
+#[test]
+fn schemas_migration_to_v2_should_work_as_expected() {
+	new_test_ext().execute_with(|| {
+		// Arrange
+		sudo_set_max_schema_size();
+		let sender: AccountId = test_public(5);
+		let schemas = vec![
+			r#"{"Name": "Bond", "Code": "007"}"#,
+			r#"{"type": "num","minimum": -90,"maximum": 90}"#,
+			r#"{"latitude": 48.858093,"longitude": 2.294694}"#,
+		];
+		for (idx, fields) in schemas.iter().enumerate() {
+			assert_ok!(SchemasPallet::create_schema_v2(
+				RuntimeOrigin::signed(sender.clone()),
+				create_bounded_schema_vec(fields),
+				ModelType::AvroBinary,
+				PayloadLocation::OnChain,
+				BoundedVec::default()
+			));
+			v2::old::Schemas::<Test>::insert(
+				idx as u16 + 1,
+				v2::old::Schema {
+					model_type: ModelType::AvroBinary,
+					payload_location: PayloadLocation::OnChain,
+					settings: SchemaSettings::all_disabled(),
+					model: BoundedVec::try_from(fields.as_bytes().to_vec())
+						.expect("should have value"),
+				},
+			);
+		}
+		let old_schema_1 = v2::old::Schemas::<Test>::get(1u16).expect("should have value");
```

Comment:
nit/non-blocking: Though I trust that they're there, this test would be improved if it verified that all three schemas were in old::Schemas.